rust/src/nasl/builtin/string/mod.rs

```diff
@@ -19,26 +19,6 @@ use std::num::ParseIntError;
 use crate::function_set;
 use crate::nasl::syntax::NaslValue;
 
-/// `Some(string)` if constructed from either a `NaslValue::String`
-/// or `NaslValue::Data`.
-struct StringOrData(String);
-
-fn bytes_to_str(bytes: &[u8]) -> String {
-    bytes.iter().map(|x| *x as char).collect::<String>()
-}
-
-impl<'a> FromNaslValue<'a> for StringOrData {
-    fn from_nasl_value(value: &'a NaslValue) -> Result<Self, FunctionErrorKind> {
-        match value {
-            NaslValue::String(string) => Ok(Self(string.clone())),
-            NaslValue::Data(buffer) => Ok(Self(bytes_to_str(buffer))),
-            _ => Err(FunctionErrorKind::WrongArgument(
-                "Expected string or byte buffer.".to_string(),
-            )),
-        }
-    }
-}
-
```

Comment:
Although this will probably not exist if we ever autoconvert from bytes to string, this is quite useful for now, so I moved it to `nasl/function/types.rs`